drivers/sensor/bosch/bmp180/bmp180.c

```diff
@@ -0,0 +1,294 @@
+	data->md = (int16_t)sys_get_be16(&buf[20]);
+
+	return 0;
+}
+
+static int bmp180_read_uncomp_temp(const struct device *dev)
+{
+	const struct bmp180_config *config = dev->config;
+	struct bmp180_data *data = dev->data;
+	uint8_t buf[2];
+
+	if (i2c_reg_write_byte_dt(&config->i2c, BMP180_REG_CTRL_MEAS, BMP180_CMD_READ_TEMP) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "start", "temperature measurement");
+		return -EIO;
+	}
+
+	k_msleep(5);
+
+	if (i2c_burst_read_dt(&config->i2c, BMP180_REG_OUT_MSB, buf, sizeof(buf)) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "read", "temperature measurement");
+		return -EIO;
+	}
+
+	data->uncomp_temp = (int16_t)sys_get_be16(&buf[0]);
+
+	return 0;
+}
+
+static int bmp180_read_uncomp_press(const struct device *dev)
+{
+	const struct bmp180_config *config = dev->config;
+	struct bmp180_data *data = dev->data;
+	uint8_t buf[3];
+	uint8_t oversampling_setting = config->oversampling;
+	/* See Table 8 from the datasheet */
+	uint8_t ctrl_reg_val = BMP180_CMD_READ_PRESS + (oversampling_setting << 6);
+
+	if (i2c_reg_write_byte_dt(&config->i2c, BMP180_REG_CTRL_MEAS, ctrl_reg_val) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "start", "pressure measurement");
+		return -EIO;
+	}
+
+	/* Sleep for x ms based on the oversampling setting */
+	switch (oversampling_setting) {
+	case BMP180_MODE_ULTRA_LOW_POWER:
+		k_msleep(5);
+		break;
+	case BMP180_MODE_STANDARD:
+		k_msleep(8);
+		break;
+	case BMP180_MODE_HIGH_RESOLUTION:
+		k_msleep(14);
+		break;
+	case BMP180_MODE_ULTRA_HIGH_RESOLUTION:
+		k_msleep(26);
+		break;
+	default:
+		LOG_ERR("Invalid oversampling setting.");
+		return -EINVAL;
+	}
+
+	/* Read the MSB, LSB, and XLSB from the sensor */
+	if (i2c_burst_read_dt(&config->i2c, BMP180_REG_OUT_MSB, buf, sizeof(buf)) < 0) {
+		LOG_ERR(ERROR_MSG_FMT, "read", "pressure measurement");
+		return -EIO;
+	}
+
+	data->uncomp_press = (int32_t)sys_get_be24(buf);
+	data->uncomp_press >>= (8 - oversampling_setting);
+
+	return 0;
+}
+
+static int bmp180_read_temp(const struct device *dev)
+{
+	struct bmp180_data *data = dev->data;
+
+	if (bmp180_read_uncomp_temp(dev) < 0) {
+		return -EIO;
+	}
+
+	int32_t x1 = (data->uncomp_temp - (int32_t)data->ac6) * ((int32_t)data->ac5) >> 15;
+	int32_t x2 = ((int32_t)data->mc << 11) / (x1 + (int32_t)data->md);
+	int32_t b5 = x1 + x2;
+
+	data->temp = (b5 + 8) >> 4;
+
+	return 0;
+}
+
+static int bmp180_read_press(const struct device *dev)
+{
+	const struct bmp180_config *config = dev->config;
+	struct bmp180_data *data = dev->data;
+
+	if (bmp180_read_uncomp_temp(dev) < 0) {
+		return -EIO;
+	}
+
+	if (bmp180_read_uncomp_press(dev) < 0) {
+		return -EIO;
+	}
+
+	/* See datasheet, page 15 for more details */
+	int32_t x1 = (data->uncomp_temp - (int32_t)data->ac6) * ((int32_t)data->ac5) >> 15;
+	int32_t x2 = ((int32_t)data->mc << 11) / (x1 + (int32_t)data->md);
+	int32_t b5 = x1 + x2;
+
+	int32_t b6 = b5 - 4000;
+
+	x1 = (data->b2 * ((b6 * b6) >> 12)) >> 11;
+	x2 = (data->ac2 * b6) >> 11;
```

Comment:
you might want to simplify from the original data-sheet example to 
x1 = ((b2 * (b6 * b6)>>12))>>11;
x1 = ((b2 * b6 * b6) / pow(2,12) )/ pow(2,11)
x1 = (b2 * b6 *b6) (pow(2,12) * pow(2,11))
x1 = (b2 * b6 * b6) / 0x800000;  << much